lib/opam.ml

```diff
@@ -1,13 +1,43 @@
 open Import
 
-let depends_on_dune ~allow_jbuilder (formula : OpamTypes.filtered_formula) =
-  let dune = OpamPackage.Name.of_string "dune" in
-  let jbuilder = OpamPackage.Name.of_string "jbuilder" in
-  let is_duneish name =
-    let eq n n' = OpamPackage.Name.compare n n' = 0 in
-    eq dune name || (allow_jbuilder && eq jbuilder name)
+let depends_on_any packages (formula : OpamTypes.filtered_formula) =
+  let packages = List.map ~f:OpamPackage.Name.of_string packages in
+  let is_one_of_packages name =
+    List.exists packages ~f:(fun p -> OpamPackage.Name.compare p name = 0)
   in
-  OpamFormula.fold_left (fun acc (name, _) -> acc || is_duneish name) false formula
+  OpamFormula.fold_left (fun acc (name, _) -> acc || is_one_of_packages name) false formula
+
+let ocaml_options =
+  (* ocaml-options-vanilla is purposefully excluded from that list as it doesn't imply
+     a dependency towards ocaml-variants on its own. *)
+  [
+    "ocaml-option-32bit";
```

Comment:
FYI, if you want to be future proof you can probably generate this list using `Ocaml_version`: these packages exist only since `Since.options_packages` and they are listed in `Opam.V2.additional_packages`.

Reply:
I looked into this but it unfortunately doesn't do what you suggest:
```ocaml
let additional_packages t =
      if Has.options_packages t then
        match Configure_options.of_t t with
        | Ok []
        | Error _ -> []
        | Ok options ->
            let options_only_package =
              List.map Configure_options.to_string options
              |> String.concat "-"
              |> (^) "ocaml-options-only-" in
            [options_only_package]
      else
        []
```

IIUC, It returns the `ocaml-option-*` package corresponding to the version, e.g. for `4.12.0+flambda+afl` it will return `ocaml-option-only-flambda-afl`.
I'd be happy to externalize the maintenance of this all logic as much as possible, eventually not even having to deal with it in opam-monorepo at all and have the solver do the right thing according to the user's dependencies spec but in the meantime we need a fix for it.